frontend/src/metabase/metadata/pages/DataModel/components/TablePicker/wrappers.tsx

```diff
@@ -9,13 +9,18 @@ import { getUrl } from "./utils";
 
 export function RouterTablePicker(props: TreePath) {
   const dispatch = useDispatch();
+  const [value, setValue] = useState(props);
   const onChange = useCallback(
     (value: TreePath) => {
+      setValue(value);
       dispatch(push(getUrl(value)));
     },
     [dispatch],
   );
-  return <TablePicker value={props} onChange={onChange} />;
+  useEffect(() => {
```

Comment:
This feels like it might cause a headache someday haha. Was there an action where pushing a new URL didn't impact the props being passed?

Reply:
There is a bug where if the navigation is cancelled somehow, the TablePicker is stuck in an intermediate state.